### PR TITLE
Add SU-6 smart pistol

### LINKED ISSUE #1132: SU-6 Smart Pistol Crate (x2) Cost: $3000

### LINKED ISSUE #1390: 15P SU-6 Smart Pistol

### LINKED ISSUE #1504: 12P SU-6 Smart Pistol

### LINKED ISSUE #1656: 12P SU-6 Smart Pistol

### LINKED ISSUE #1773: 12P SU-6 Smart Pistol

### LINKED ISSUE #1913: 15P SU-6 Smart Pistol

### PR DESCRIPTION
**Pull request by Dutch-VanDerLinde**
## Why / Balance
CM13 content
fixes #1390
fixes #1132
fixes #1504
fixes #1656
fixes #1773
fixes #1913

## Media

https://github.com/user-attachments/assets/c8e1e993-1821-45c2-b2d5-ee755b2cb9c4

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**
:cl:
- add: Added the SU-6 smart pistol and gun case

### CODE REVIEW

File: Resources/Prototypes/_RMC14/Entities/Objects/Weapons/gun_cases.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  - type: Appearance
+  - type: GenericVisualizer
+    visuals:
+      enum.StorageVisuals.Open:
+        base:
+          True: { state: open }
+          False: { state: closed }
+        sniper:
+          True: { visible: false }
+          False: { visible: true }
```
Comment (QuietlyWhisper) [inline Resources/Prototypes/_RMC14/Entities/Objects/Weapons/gun_cases.yml:26]:
```suggestion
```
This isn't referenced/used anywhere, remove it

File: Resources/Prototypes/_RMC14/Catalog/Fills/Crates/magazine_boxes.yml [hunk 2]
```diff
@@ -141,3 +141,13 @@
...
+
+- type: entity
+  parent: CMCrateAmmo
+  id: RMCCrateMagazineSmartPistol
+  name: SU-6 smart pistol magazine crate (8x)
+  components:
+  - type: StorageFill
+    contents:
+    - id: RMCMagazinePistolSU6
+      amount: 8
```
Comment (QuietlyWhisper) [inline Resources/Prototypes/_RMC14/Catalog/Fills/Crates/magazine_boxes.yml:153]: This should be a crate of two magazine box and not a crate of mags

File: Resources/Prototypes/_RMC14/Catalog/Cargo/requisitions_catalog.yml [hunk 3]
```diff
@@ -40,6 +40,8 @@
         crate: RMCCrateBatterySmartGun
       - cost: 3000
         crate: RMCCrateMagazineSmartGun
+      - cost: 4000
+        crate: RMCCrateMagazineSmartPistol
```
Comment (QuietlyWhisper) [inline Resources/Prototypes/_RMC14/Catalog/Cargo/requisitions_catalog.yml:44]: It's supposed to be a crate of 2 magazine boxes, not a crate of magazines

File: Resources/Prototypes/_RMC14/Entities/Clothing/Belt/belts.yml [hunk 4]
```diff
@@ -309,3 +309,26 @@
...
+  description: The M276 is the standard load-bearing equipment of the UNMC. It consists of a modular belt with various clips. This version is for the SU-6 smartpistol.
+  components:
+  - type: Sprite
+    sprite: _RMC14/Objects/Clothing/Belt/smart_gun_operator_pistol.rsi
+  - type: Clothing
+    sprite: _RMC14/Objects/Clothing/Belt/smart_gun_operator_pistol.rsi
+  - type: Storage
+    maxItemSize: Normal
+    grid:
+    - 0,0,11,1 # 6 slots
```
Comment (QuietlyWhisper) [inline Resources/Prototypes/_RMC14/Entities/Clothing/Belt/belts.yml:326]: Should be 7 slots
```suggestion
    grid:
    - 0,0,13,1 # 7 slots
```

File: Resources/Prototypes/_RMC14/Entities/Structures/Machines/Vending/Crew/requisitions.yml [hunk 5]
```diff
@@ -39,8 +39,8 @@
       entries:
       - id: CMWeaponPistolMK80
         amount: 4
-      #- id: SU-6 Smart Pistol
-      #  amount: 3
+      - id: RMCWeaponPistolSU6
```
Comment (QuietlyWhisper) [inline Resources/Prototypes/_RMC14/Entities/Structures/Machines/Vending/Crew/requisitions.yml:42]: This is supposed to be three pistol cases that include the belt
```suggestion
      - id: RMCSmartPistolCase
```

File: Resources/Prototypes/_RMC14/Entities/Objects/Weapons/gun_cases.yml [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    - 0,0,3,1
+    whitelist:
+      tags:
+      - RMCSmartPistol
+      - RMCMagazinePistolSU6
+  - type: StorageFill
+    contents:
+    - id: RMCWeaponPistolSU6
+    - id: RMCMagazinePistolSU6
+      amount: 2
```
Comment (QuietlyWhisper) [inline Resources/Prototypes/_RMC14/Entities/Objects/Weapons/gun_cases.yml:73]: I don't see this used anywhere nor referenced in 13
```suggestion
```

**Review by QuietlyWhisper — CHANGES_REQUESTED** [review]
You're missing actual magazine boxes and their crafting recipies,
look at usages of  `CMBoxMagazinePistolM1984` for all the places it's used for examples

There's also a few entities in this pr that don't exist, make sure everything is actually something that comes from 13

**Review by QuietlyWhisper — DISMISSED** [review]
Seems okay, I'll test it in game but it seems good to go otherwise
Thanks

**Review by DrSmugleaf — APPROVED** [review]
Thanks for the PR!

**Pull request merged**